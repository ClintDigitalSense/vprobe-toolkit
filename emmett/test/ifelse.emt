SimpleIf if (1) b = 19;
SimpleIfElse if (2) aWasTrue = 1; else aWasTrue = 0;

# nested, without braces. "else" should stick to the most recent "if"
NestedIfElseNoBrace
if (3)
 	outer = 1;
else if (4)
	innerTrue = 1;
else innerFalse = 1;

QuestionMarkColon
        logint(VCPUID == 0 ? 0 : 12);
