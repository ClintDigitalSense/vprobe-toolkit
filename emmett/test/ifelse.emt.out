(version 0.2)
(definteger aWasTrue)
(definteger b)
(definteger innerFalse)
(definteger innerTrue)
(definteger outer)

 ;;;;;
(vprobe SimpleIf ( cond  (1( setint b   19))))
 ;;;;;
(vprobe SimpleIfElse ( cond  (2( setint aWasTrue   1)) (1 ( setint aWasTrue   0))))
 ;;;;;
(vprobe NestedIfElseNoBrace ( cond  (3( setint outer   1)) (1 ( cond  (4( setint innerTrue   1)) (1 ( setint innerFalse   1))))))
 ;;;;;
(vprobe QuestionMarkColon ( logint    (cond ( (== VCPUID  0 ) 0)(1 12)) ))